unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -507,6 +507,48 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
     }
     return res;
   }
+  
+  private class Code {
+    public final Character value;
+    public Code(Character value) {
+      this.value = value;
+    }
+    public boolean isEmpty(){
+      return value == null;
+    }
+  }
+
+  public UTF8String translate(UTF8String matchingString, UTF8String replaceString) {
+    String srcStr = this.toString();
+    String matching = matchingString.toString();
+    String replace = replaceString.toString();
+    Map<Character, Code> dict = new HashMap<>();
```

Comment:
Had that idea before but just in case we would pass `0` within the string. If that's acceptable I would update the code shortly.